### PR TITLE
Pass on captured output from acsxxx.e and calacs.e 

### LINKED ISSUE #42: Pass subprocess.call() results to user
In acstools, all of the ``subprocess.call()`` that execute the underlying HSTCAL C executables, never return stdout nor stderr. When the called exe fails it does so silently.

For example, [ascrej()](https://github.com/spacetelescope/acstools/blob/master/acstools/acsrej.py#L187)
```Python
...
  if newbias:
        call_list.append('-newbias')

    if exe_args:
        call_list.extend(exe_args)

    subprocess.call(call_list) # <---------------------------- THIS LINE


def getHelpAsString():
    """
...
```

Perhaps we should bump this up to using ``subprocess.run()``, e.g.
```python
results = subprocess.run(cmd, shell=True, check=False, universal_newlines=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

__or__ ``subprocess.check_output()``

__or__ both e.g. something like this

```pyhton
try:
        results = subprocess.run(cmd, shell=True, check=False, universal_newlines=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        if results.returncode:
            return results.stdout.rstrip()
    except AttributeError:
        try:
            results = subprocess.check_output(cmd, shell=True, universal_newlines=True, stderr=subprocess.PIPE)
            return str(results).rstrip()
        except subprocess.CalledProcessError:
            return None
    except:
        return None
```
__or__ just trivially:
```pyhton
retcode = subprocess.call(call_list)
if retcode:
    print('ERROR: return code:', retcode) 
```
__or__ better still:
```pyhton
subprocess.check_call(call_list)
```
__or__ the probably preferred option:
```python
   try:
        subprocess.check_call(call_list)
    except subprocess.CalledProcessError as err:
        if err.returncode < 0:
            raise err
        else:
            return err.returncode
    except:
        raise
    return 0
```

### PR DESCRIPTION
**Pull request by pllim**
Fix #42 

Does this solve your problem, @dborncamp, @nmiles2718, and @tddesjardins ?

**Instructions to install this branch for testing**

1. Create a directory to store the source codes and `cd` into it. (If you have done it before for another PR, just `cd` into it your existing directory.)

2. `git clone https://github.com/pllim/acstools.git ./` (this makes a copy of my ACSTOOLS fork on your disk, skip this if you already cloned it before for another PR)

3. `git fetch origin pass-on-errmsg`

4. `git checkout origin/pass-on-errmsg -b pass-on-errmsg` (if this does not work, your `git` version is too old, so you will need to update `git` first or install a newer version using `conda install git`)

5. `git status` (if this shows that you are now in `pass-on-errmsg` branch, then proceed)

6. `python setup.py install` (make sure you are in the correct conda environment first)

7. Now `cd` into your working directory with data files.

8. Use ACSTOOLS as usual in your Python session. For sanity check, you can look at the outputs of `acstools.acsccd.__version__` and `acstools.acsccd.__vdate__` and see if the match the values changed in this PR.

**Instructions to grab updates from existing branch checkout**

1. `cd` into source checkout directory.

2. `git checkout pass-on-errmsg` to switch to this branch, if not already on it (you can check with `git status`)

3. `git fetch origin pass-on-errmsg` (picks up updates)

4. `git rebase origin/pass-on-errmsg` (applies the updates locally)

5. Continue from Step 6 above.

### CODE REVIEW

**Comment by nmiles2718** [comment]
**ACSREJ TEST**
Alright, I tested this using the version of acsrej.e installed under https://github.com/spacetelescope/hstcal/pull/2 and the current version in the last release of hstcal (v1.3.0). In the current version of hstcal, I could not get the error to occur as it handled up to 120 files no problem and there is a hard limit of only 120 input files being allowed in that version acsrej.

In the newer version installed with https://github.com/spacetelescope/hstcal/pull/2, the traceback was indeed displayed showing a segfault. The 2.1 printed out first is the result of (acstools.acsccd.__version__) which ensured that this version of acstools was being used (see step 8) above) and the git output is from the updated version of acsrej installed with https://github.com/spacetelescope/hstcal/pull/2. I ran this on 120 files, but I removed some of the `ERROR:     Cant open file _____ ` lines for brevity to highlight that the traceback was indeed handled properly and displayed.
```(acstools_errmsg) nmiles:/user/nmiles/test_acsrej_ticket976/code$python run_acsrej.py 
2.1
git tag: UNKNOWN
git branch: lim-calacs-ticket976
HEAD @: 331d591e8de487be2071897ace10291f390eaa59
ERROR:    Can't open file /Users/nmiles/acstools_test_data/j8jcowrdq_flt.tra
ERROR:    Can't open file /Users/nmiles/acstools_test_data/j8jcoxrnq_flt.tra
ERROR:    Can't open file /Users/nmiles/acstools_test_data/j8jcoytlq_flt.tra
ERROR:    Can't open file /Users/nmiles/acstools_test_data/j8jcpccsq_flt.tra
.
.
.
ERROR:    Can't open file /Users/nmiles/acstools_test_data/j8jefkt5q_flt.tra
ERROR:    Can't open file /Users/nmiles/acstools_test_data/j8jefluyq_flt.tra
ERROR:    Can't open file /Users/nmiles/acstools_test_data/j8jefmvsq_flt.tra
ERROR:    Can't open file /Users/nmiles/acstools_test_data/j8jefnwmq_flt.tra


CALBEG*** ACSREJ -- Version 9.2.0 (01-Jun-2017) ***
Begin    07-Nov-2017 15:10:34 EST


CRCORR   PERFORM
Trying to open /Users/nmiles/acstools_test_data/j8jcowrdq_flt.fits...
Read in Primary header from /Users/nmiles/acstools_test_data/j8jcowrdq_flt.fits...

subprocess return code: -11 (segfault)
```

**Comment by pllim** [comment]
From what @nmiles2718 reported, this PR seems to fix the issue. However, I cannot merge without someone approving the PR and GitHub is smart enough to not let me approve my own PR and I am too lazy to have two GitHub accounts. Someone please hit approve and merge. Thanks.

**Comment by dborncamp** [comment]
I am not authorized...

**Comment by tddesjardins** [comment]
Checked and I can report the same. Not authorized.

________________________________

**Comment by pllim** [comment]
No worries. Thanks for trying!

**Comment by jamienoss** [comment]
>  However, I cannot merge without someone approving the PR and GitHub is smart enough to not let me approve my own PR and I am too lazy to have two GitHub accounts. Someone please hit approve and merge. Thanks.

I'm not really sure what to say here other than NO, NO, and NO. There are official reviewers requested so let that be the case. Let us take no further attempt to sidestep this process, no matter how trivial this change may seem.

I looked at this yesterday and didn't approve because there were things that I didn't like. When I finish the review you'll know.

File: acstools/acs2d.py [hunk 1]
```diff
@@ -110,7 +115,19 @@ def acs2d(input, exec_path='', time_stamps=False, verbose=False, quiet=False, ex
     if exe_args:
         call_list.extend(exe_args)
 
-    subprocess.call(call_list)
+    # Piping out to subprocess.PIPE or sys.stdout don't seem to work here.
+    with tempfile.TemporaryFile() as fp:
+        retcode = subprocess.call(call_list, stdout=fp, stderr=fp)
```
[the remark this answers is not in the document]
Comment (jamienoss) [reply acstools/acs2d.py]: Actually, it may even be better to return `retcode` instead of just printing it.

**Comment by pllim** [comment]
Update: I discussed this with @jamienoss in detail and we both agreed that he would open an alternate pull request on how else this problem could be fixed.

**Comment by jamienoss** [comment]
@pllim 
>Update: I discussed this with @jamienoss in detail and we both agreed that he would open an alternate pull request on how else this problem could be fixed.

My alternative suggestions are #48 & #49 

The former will raise an exception even if the called exec handles/creates an error but still returns a non-zero return code, i.e. 2 for `ERROR_RETURN` ([source code](https://github.com/spacetelescope/hstcal/blob/master/include/hstcalerr.h#L12)).
The latter will not, and only raises if the return code is negative, which is normally the case for OS signal classes, e.g. `SIGSEG`, `SIGABRT`, etc.

**Comment by pllim** [comment]
Closing in favor of #48.

**Pull request closed without merging**